Sources/Visitor.swift

```diff
@@ -21,51 +21,144 @@ import Foundation
 
 // MARK: - Visitor Protocol
 
+/**
+ The [Visitor Design Pattern](https://en.wikipedia.org/wiki/Visitor_pattern) is
+ used throughout the MiniDOM library to implement algorithms that involve 
+ traversing the DOM tree. It provides a convenient mechanism to separate an 
+ algorithm from the object structure on which it operates. It allows operations
+ to be added to the DOM structure without modifying the structures themselves.
+ 
+ A `Visitor` object is provided to the `Node.accept(_:)` to start the traversal.
+ The `Node` object calls the appropriate methods on the `Visitor` object before
+ traversing recursively through the child nodes.
```

Comment:
before calling `Node.accept(_:)` on its child nodes, performing the recursive traversal.